Fix test suite

Our test suite has several problems.

1. `bluemix` generator is calling `process.exit()` on failure, which aborts Mocha test suite in the middle. Because the process exits with exit code `0`, it looks like all tests passed.

    https://github.com/strongloop/generator-loopback/blob/060bdd7dd0deb17e1df0b58a36843d82d4bc953d/bluemix/index.js#L74-L75

    I am proposing to change this block and throw an Error instead.

    ```js
    throw new Error('Invalid LoopBack directory');
    ```

2. The test suite for `bluemix` generator works when executed on its own (`mocha test/bluemix.test.js`) passes, but fails when executed together with other tests. We need to find and fix the reason.

3. Setting the bluemix test suite aside, several other tests are failing. These tests are not executed now because bluemix exits the process prematurely. We need to fix these tests to pass again.

At my machine, there are 17 failing tests.

  1) loopback:bluemix generator should generate datasources.bluemix.js:
  2) loopback:bluemix generator should generate .bluemix/datasources-config.json:
  3) loopback:bluemix generator should generate all basic Bluemix files:
  4) loopback:bluemix generator should generate only basic Bluemix files:
  5) loopback:bluemix generator should generate only Docker files:
  6) loopback:bluemix generator should generate only toolchain files:
  7) loopback:bluemix generator should generate manifest file:
  8) loopback:export-api-def generator produces JSON format:
  9) loopback:export-api-def generator produces YAML format:
  10) loopback:export-api-def generator running on project with onging async operations 
  11) loopback:export-api-def generator running on project with onging async operations eventually exits the process:
  12) loopback generator help prints right help message for each generator prints right help message for generator app:
  13) loopback:middleware generator adds a new phase to server/middleware.json:
  14) loopback:middleware generator adds a new phase next to a selected one:
  15) loopback:middleware generator adds a new entry to an existing phase in server/middleware.json:
  16) loopback:middleware generator supports sub-phase:
  17) loopback:soap tests periodic table wsdl periodic table wsdl:
     Uncaught Error: Invalid WSDL URL: http://www.webservicex.net/periodictable.asmx?WSDL

By looking at the detailed logs (see below), I suspect most test are failing because of the same reason, which is that somehow loopback-workspace is creating a new project in `test/sandbox/test-app` instead of `test/sandbox`, where the tests expect the project to be created.

>   12) loopback generator help prints right help message for each generator prints right help message for generator app:

This test should be easy to fix, just edit expected help text to accommodate recent additions.

>   17) loopback:soap tests periodic table wsdl periodic table wsdl:
     Uncaught Error: Invalid WSDL URL: http://www.webservicex.net/periodictable.asmx?WSDL

Looks like the `periodictable` API is gone. I think we will need to rework the test to use a different api, see http://www.webservicex.net/api

<details>
<summary>Expand to see full failure details</summary>
<p>
  1) loopback:bluemix generator should generate datasources.bluemix.js:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  2) loopback:bluemix generator should generate .bluemix/datasources-config.json:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  3) loopback:bluemix generator should generate all basic Bluemix files:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  4) loopback:bluemix generator should generate only basic Bluemix files:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  5) loopback:bluemix generator should generate only Docker files:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  6) loopback:bluemix generator should generate only toolchain files:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  7) loopback:bluemix generator should generate manifest file:
     Uncaught Error: Invalid LoopBack directory
      at constructor.validateLoopBackDir (bluemix/index.js:74:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  8) loopback:export-api-def generator produces JSON format:
     Uncaught Error: Cannot find module '/Users/bajtos/src/loopback/generator-loopback/test/sandbox'
      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
      at Function.Module._load (internal/modules/cjs/loader.js:520:25)
      at Module.require (internal/modules/cjs/loader.js:650:17)
      at require (internal/modules/cjs/helpers.js:20:18)
      at constructor.generate (export-api-def/index.js:38:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  9) loopback:export-api-def generator produces YAML format:
     Uncaught Error: Cannot find module '/Users/bajtos/src/loopback/generator-loopback/test/sandbox'
      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
      at Function.Module._load (internal/modules/cjs/loader.js:520:25)
      at Module.require (internal/modules/cjs/loader.js:650:17)
      at require (internal/modules/cjs/helpers.js:20:18)
      at constructor.generate (export-api-def/index.js:38:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  10) loopback:export-api-def generator running on project with onging async operations immediately does not exit the process:
     Uncaught Error: Cannot find module '/Users/bajtos/src/loopback/generator-loopback/test/sandbox'
      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
      at Function.Module._load (internal/modules/cjs/loader.js:520:25)
      at Module.require (internal/modules/cjs/loader.js:650:17)
      at require (internal/modules/cjs/helpers.js:20:18)
      at constructor.generate (export-api-def/index.js:38:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  11) loopback:export-api-def generator running on project with onging async operations eventually exits the process:
     Uncaught Error: Cannot find module '/Users/bajtos/src/loopback/generator-loopback/test/sandbox'
      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
      at Function.Module._load (internal/modules/cjs/loader.js:520:25)
      at Module.require (internal/modules/cjs/loader.js:650:17)
      at require (internal/modules/cjs/helpers.js:20:18)
      at constructor.generate (export-api-def/index.js:38:15)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/base.js:439:23)
      at node_modules/run-async/index.js:25:25
      at new Promise (<anonymous>)
      at node_modules/run-async/index.js:24:19
      at node_modules/yeoman-generator/lib/base.js:440:9
      at process.topLevelDomainCallback (domain.js:121:23)

  12) loopback generator help prints right help message for each generator prints right help message for generator app:

      AssertionError [ERR_ASSERTION]: 'Usage:\n  slc loopback:app [options] [<name>]\n\nOptions:\n  -h,   --help             # Print the command\'s options and usa... == 'Usage:\n  slc loopback:app [options] [<name>]\n\nOptions:\n  -h,   --help             # Print the command\'s options and usa...
      + expected - actual

               --skip-cache       # Do not remember prompt answers                          Default: false
               --skip-install     # Do not install npm dependencies                         Default: false
               --skip-next-steps  # Do not print "next steps" info
               --explorer         # Add Loopback Explorer to the project (true by default)
      +        --loopbackVersion  # Select the LoopBack version
      +        --template         # Set up the LoopBack application template
               --bluemix          # Set up as a Bluemix app

       Arguments:
         name  # Name of the application to scaffold.  Type: String  Required: false

      at Context.<anonymous> (test/help.test.js:83:16)
      at process.topLevelDomainCallback (domain.js:121:23)

  13) loopback:middleware generator adds a new phase to server/middleware.json:
     Error: ENOENT: no such file or directory, open '/Users/bajtos/src/loopback/generator-loopback/test/sandbox/server/middleware.json'
      at Object.openSync (fs.js:443:3)
      at Object.readFileSync (fs.js:348:35)
      at readMiddlewaresJsonSync (test/middleware.test.js:113:22)
      at Context.<anonymous> (test/middleware.test.js:34:38)
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:726:13
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:52:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:269:32
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:44:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:723:17
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:167:37
      at process._tickCallback (internal/process/next_tick.js:61:11)

  14) loopback:middleware generator adds a new phase next to a selected one:

      Uncaught AssertionError: expected 0 to equal -2
      + expected - actual

      -0
      +-2

      at test/middleware.test.js:58:21
      at constructor.<anonymous> (node_modules/yeoman-generator/lib/base.js:419:5)
      at Queue.EventEmitter.emit (domain.js:442:20)
      at Queue.<anonymous> (node_modules/grouped-queue/lib/queue.js:68:12)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:79:43)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at SubQueue.run (node_modules/grouped-queue/lib/subqueue.js:42:45)
      at Queue.next (node_modules/grouped-queue/lib/queue.js:80:39)
      at Queue._exec (node_modules/grouped-queue/lib/queue.js:83:3)
      at constructor.<anonymous> (node_modules/yeoman-generator/lib/base.js:495:7)
      at node_modules/async/dist/async.js:3888:9
      at node_modules/async/dist/async.js:473:16
      at replenish (node_modules/async/dist/async.js:1006:25)
      at node_modules/async/dist/async.js:1016:9
      at eachOfLimit (node_modules/async/dist/async.js:1041:24)
      at node_modules/async/dist/async.js:1046:16
      at _parallel (node_modules/async/dist/async.js:3879:5)
      at Object.series (node_modules/async/dist/async.js:4735:5)
      at Conflicter.resolve (node_modules/yeoman-generator/lib/util/conflicter.js:91:9)
      at constructor.<anonymous> (node_modules/yeoman-generator/lib/base.js:490:21)
      at process.topLevelDomainCallback (domain.js:121:23)

  15) loopback:middleware generator adds a new entry to an existing phase in server/middleware.json:
     Error: ENOENT: no such file or directory, open '/Users/bajtos/src/loopback/generator-loopback/test/sandbox/server/middleware.json'
      at Object.openSync (fs.js:443:3)
      at Object.readFileSync (fs.js:348:35)
      at readMiddlewaresJsonSync (test/middleware.test.js:113:22)
      at Context.<anonymous> (test/middleware.test.js:74:9)
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:726:13
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:52:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:269:32
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:44:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:723:17
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:167:37
      at process._tickCallback (internal/process/next_tick.js:61:11)

  16) loopback:middleware generator supports sub-phase:
     Error: ENOENT: no such file or directory, open '/Users/bajtos/src/loopback/generator-loopback/test/sandbox/server/middleware.json'
      at Object.openSync (fs.js:443:3)
      at Object.readFileSync (fs.js:348:35)
      at readMiddlewaresJsonSync (test/middleware.test.js:113:22)
      at Context.<anonymous> (test/middleware.test.js:94:7)
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:726:13
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:52:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:269:32
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:44:16
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:723:17
      at node_modules/loopback-workspace/node_modules/async/lib/async.js:167:37
      at process._tickCallback (internal/process/next_tick.js:61:11)

  17) loopback:soap tests periodic table wsdl periodic table wsdl:
     Uncaught Error: Invalid WSDL URL: http://www.webservicex.net/periodictable.asmx?WSDL

 Code: 404

 Response Body: "{\"code\":\"PAGE_NOT_FOUND\",\"message\":\"Page not found\"}\n"
      at node_modules/strong-soap/src/parser/wsdl.js:419:22
      at Request._callback (node_modules/strong-soap/src/http.js:136:9)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.EventEmitter.emit (domain.js:442:20)
      at Request.<anonymous> (node_modules/request/request.js:1157:10)
      at Request.EventEmitter.emit (domain.js:442:20)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1079:12)
      at IncomingMessage.EventEmitter.emit (domain.js:442:20)
      at endReadableNT (_stream_readable.js:1081:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)
</p>
</details>

--- a/app/index.js
+++ b/app/index.js
@@ -252,8 +252,8 @@ module.exports = yeoman.Base.extend({
         cb();
       } else {
         cb(new Error(
-          g.f('The generator must be run in an empty directory.'))
-        );
+          g.f('The generator must be run in an empty directory.')
+        ));
       }
     });
   },
@@ -349,7 +349,8 @@ module.exports = yeoman.Base.extend({
           'support and maintain LoopBack, which is at the core of\n' +
           'API Connect. When your APIs need robust management and\n' +
           'security options, please check out %s',
-        'http://ibm.biz/tryAPIC')));
+        'http://ibm.biz/tryAPIC'
+      )));
       this.log();
     },
   },

--- a/bluemix/index.js
+++ b/bluemix/index.js
@@ -69,10 +69,11 @@ module.exports = yeoman.Base.extend({
   validateLoopBackDir: function() {
     var root = this.destinationRoot();
     if (!(this.options.login || this.options.sso)) {
-      if (!fs.existsSync(path.join(root, 'package.json')) ||
-        !fs.existsSync(path.join(root, 'server', 'server.js'))) {
-        this.log(chalk.red('\n Invalid LoopBack directory\n'));
-        process.exit();
+      if (
+        !fs.existsSync(path.join(root, 'package.json')) ||
+        !fs.existsSync(path.join(root, 'server', 'server.js'))
+      ) {
+        throw new Error('Invalid LoopBack directory');
       }
     }
   },

--- a/bluemix/login.js
+++ b/bluemix/login.js
@@ -108,7 +108,8 @@ function login() {
           });
           if (choices.length === 1) {
             this.log(chalk.cyan(
-              g.f('> Default organization: ') + choices[0].name));
+              g.f('> Default organization: ') + choices[0].name
+            ));
             this.organization = choices[0].value;
           }
           var prompts = [
@@ -136,7 +137,8 @@ function login() {
                 });
                 if (choices.length === 1) {
                   this.log(chalk.cyan(
-                    g.f('> Default space: ') + choices[0].name));
+                    g.f('> Default space: ') + choices[0].name
+                  ));
                   this.space = choices[0].value;
                 }
                 var prompts = [

--- a/datasource/index.js
+++ b/datasource/index.js
@@ -253,7 +253,8 @@ module.exports = yeoman.Base.extend({
       npmModule += '@' + pkg.version;
     }
     var projectPkg = JSON.parse(
-      fs.readFileSync(path.join(this.projectDir, 'package.json'), 'utf-8'));
+      fs.readFileSync(path.join(this.projectDir, 'package.json'), 'utf-8')
+    );
     if (projectPkg.dependencies[pkg.name]) {
       return;
     }

--- a/middleware/index.js
+++ b/middleware/index.js
@@ -224,7 +224,8 @@ module.exports = yeoman.Base.extend({
       helpers.reportValidationError(err, self.log);
       if (!err && inst) {
         self.log(g.f(
-          'Middleware %s is added to phase %s.', inst.name, inst.phase));
+          'Middleware %s is added to phase %s.', inst.name, inst.phase
+        ));
       }
       return done(err);
     });

--- a/model/index.js
+++ b/model/index.js
@@ -186,7 +186,8 @@ module.exports = yeoman.Base.extend({
         if (err) return done(err);
         this.baseModel = model;
         done();
-      }.bind(this));
+      }.bind(this)
+    );
   },
 
   askForParameters: function() {
@@ -355,7 +356,8 @@ module.exports = yeoman.Base.extend({
             this.log(g.f('\nLet\'s add another %s property.',
               this.displayName));
             this.property();
-          }.bind(this));
+          }.bind(this)
+        );
       }.bind(this));
     }
   },

--- a/oracle/index.js
+++ b/oracle/index.js
@@ -67,14 +67,16 @@ module.exports = yeoman.Base.extend({
   checkConnector: function() {
     try {
       var m = require(
-        this.destinationPath('node_modules/loopback-connector-oracle'));
+        this.destinationPath('node_modules/loopback-connector-oracle')
+      );
       this.log(chalk.green(g.f('Oracle connector is ready.')));
       if (!this.options.connector && !this.options.driver) this.skip = true;
     } catch (e) {
       if (e.code === 'MODULE_NOT_FOUND') {
         // loopback-connector-oracle is not installed
         this.log(chalk.red(
-          g.f('Module is not installed: ' + e)));
+          g.f('Module is not installed: ' + e)
+        ));
         this.options.connector =
           e.message.indexOf('loopback-connector-oracle') !== -1;
         this.options.driver =
@@ -82,7 +84,8 @@ module.exports = yeoman.Base.extend({
       } else {
         // oracledb cannot be loaded due to dynamic lib issues
         this.log(chalk.red(
-          g.f('Module oracledb fails to load: ' + e)));
+          g.f('Module oracledb fails to load: ' + e)
+        ));
         this.options.driver = true;
       }
     }
@@ -151,7 +154,8 @@ module.exports = yeoman.Base.extend({
         // Delete node_modules/loopback-oracle-installer so that
         // npm install loopback-connector-oracle will trigger post-install
         var dir = this.destinationPath(
-          'node_modules/loopback-oracle-installer');
+          'node_modules/loopback-oracle-installer'
+        );
         if (this.options.verbose) {
           this.log('Removing directory: ' + dir);
         }
@@ -186,7 +190,8 @@ module.exports = yeoman.Base.extend({
         // so that the top-level oracledb will be used
         var dir =
           this.destinationPath(
-            'node_modules/loopback-connector-oracle/node_modules/oracledb');
+            'node_modules/loopback-connector-oracle/node_modules/oracledb'
+          );
         if (this.options.verbose) {
           this.log('Removing directory: ' + dir);
         }
@@ -211,20 +216,24 @@ module.exports = yeoman.Base.extend({
         // here as it's cached by previous require
         // First check node_modules/oracledb
         require(
-          this.destinationPath('node_modules/oracledb'));
+          this.destinationPath('node_modules/oracledb')
+        );
         this.log(chalk.green(g.f('Oracle driver is ready.')));
       } catch (e) {
         try {
           // Try the local oracledb inside loopback-connector-oracle
           require(
             this.destinationPath(
-              'node_modules/loopback-connector-oracle/node_modules/oracledb'));
+              'node_modules/loopback-connector-oracle/node_modules/oracledb'
+            )
+          );
           this.log(chalk.green(g.f('Oracle driver is ready.')));
         } catch (e) {
           this.log(chalk.red(g.f('Oracle driver fails to load: ' + e)));
           this.log(
             chalk.red(g.f('Please try `lb oracle --driver` or ' +
-              'follow instructions at ' + INSTALL_URL + '.')));
+              'follow instructions at ' + INSTALL_URL + '.'))
+          );
           var done = this.async();
 
           var prompts = [

--- a/package.json
+++ b/package.json
@@ -54,8 +54,8 @@
   "devDependencies": {
     "bluebird": "^3.1.1",
     "chai": "^3.2.0",
-    "eslint": "^4.19.1",
-    "eslint-config-loopback": "^10.0.0",
+    "eslint": "^5.2.0",
+    "eslint-config-loopback": "^11.0.0",
     "mocha": "^3.2.0",
     "nock": "^9.0.14",
     "rimraf": "^2.6.1",

--- a/remote-method/index.js
+++ b/remote-method/index.js
@@ -177,7 +177,8 @@ module.exports = yeoman.Base.extend({
           verb: answers.httpVerb || answers.customHttpVerb,
         });
         this.log(g.f(
-          '\nLet\'s add another endpoint, enter an empty name when done.'));
+          '\nLet\'s add another endpoint, enter an empty name when done.'
+        ));
         this.askForEndpoints();
       }.bind(this));
     }.bind(this));
@@ -343,7 +344,8 @@ module.exports = yeoman.Base.extend({
           description: answers.returnsDes,
         });
         this.log(g.f(
-          '\nLet\'s add another return argument. Enter empty name when done.'));
+          '\nLet\'s add another return argument. Enter empty name when done.'
+        ));
         this.askForReturns();
       }.bind(this));
     }.bind(this));

--- a/soap/index.js
+++ b/soap/index.js
@@ -369,7 +369,8 @@ module.exports = yeoman.Base.extend({
       if (!err) {
         self.log(
           chalk.green(g.f('Models are successfully generated from ' +
-            '{{WSDL}}.')));
+            '{{WSDL}}.'))
+        );
       }
       helpers.reportValidationError(err, self.log);
       done(err);

--- a/swagger/index.js
+++ b/swagger/index.js
@@ -135,11 +135,13 @@ module.exports = yeoman.Base.extend({
       function(done) {
         // Find existing model definitions
         wsModels.ModelDefinition.find(
-          {where: {name: {inq: self.modelNames}}}, done);
+          {where: {name: {inq: self.modelNames}}}, done
+        );
       },
       function(done) {
         wsModels.ModelConfig.find(
-          {where: {name: {inq: self.modelNames}}}, done);
+          {where: {name: {inq: self.modelNames}}}, done
+        );
       }],
     function(err, objs) {
       if (err) {
@@ -347,7 +349,8 @@ module.exports = yeoman.Base.extend({
       if (!err) {
         self.log(
           chalk.green(g.f('Models are successfully generated from ' +
-            '{{swagger spec}}.')));
+            '{{swagger spec}}.'))
+        );
       }
       helpers.reportValidationError(err, self.log);
       done(err);

--- a/test/acl.test.js
+++ b/test/acl.test.js
@@ -41,8 +41,10 @@ describe('loopback:acl generator', function() {
           {
             name: 'Location',
             facetName: 'common',
-          }, done);
-      });
+          }, done
+        );
+      }
+    );
   });
 
   it('adds an entry to models.json', function(done) {

--- a/test/app.test.js
+++ b/test/app.test.js
@@ -214,7 +214,12 @@ describe('loopback:app generator', function() {
       gen.run(function(err) {
         expect(err.message).to.eql(
           'Invalid LoopBack version: invalid-version. ' +
-          'Available versions are 2.x, 3.x.');
+          'Available versions are 2.x, 3.x.'
+        );
+
+        // a way to get around Workspace.copyRecursive not being properly
+        // restored to its original method
+        gen.emit('end');
         done();
       });
     });
@@ -235,7 +240,12 @@ describe('loopback:app generator', function() {
       gen.run(function(err) {
         expect(err.message).to.eql(
           'Invalid template: invalid-template. Available templates for 3.x ' +
-          'are api-server, empty-server, hello-world, notes');
+          'are api-server, empty-server, hello-world, notes'
+        );
+
+        // a way to get around Workspace.copyRecursive not being properly
+        // restored to its original method
+        gen.emit('end');
         done();
       });
     });

--- a/test/bluemix.test.js
+++ b/test/bluemix.test.js
@@ -187,7 +187,8 @@ describe('loopback:bluemix generator', function() {
         assert.equal('string', typeof config.space.name);
         done();
       });
-    });
+    }
+  );
 
   itSkipIf(!process.env.BLUEMIX_EMAIL || !process.env.BLUEMIX_PASSWORD)(
     'should login with SSO passcode', function(done) {
@@ -211,7 +212,8 @@ describe('loopback:bluemix generator', function() {
         assert.equal('string', typeof config.space.name);
         done();
       });
-    });
+    }
+  );
 });
 
 function givenBluemixGenerator(args) {

--- a/test/datasource.test.js
+++ b/test/datasource.test.js
@@ -73,7 +73,8 @@ describe('loopback:datasource generator', function() {
 
     datasourceGen.run(function() {
       var pkg = fs.readFileSync(
-        path.join(SANDBOX, 'package.json'), 'UTF-8');
+        path.join(SANDBOX, 'package.json'), 'UTF-8'
+      );
       pkg = JSON.parse(pkg);
       // eslint-disable-next-line no-unused-expressions
       expect(pkg.dependencies['loopback-connector-rest']).to.exist;
@@ -91,7 +92,8 @@ describe('loopback:datasource generator', function() {
 
     datasourceGen.run(function() {
       var pkg = fs.readFileSync(
-        path.join(SANDBOX, 'package.json'), 'UTF-8');
+        path.join(SANDBOX, 'package.json'), 'UTF-8'
+      );
       pkg = JSON.parse(pkg);
       // eslint-disable-next-line no-unused-expressions
       expect(pkg.dependencies.lodash).to.exist;
@@ -274,7 +276,8 @@ describe('loopback:datasource generator', function() {
             var datasources = Object.keys(readDataSourcesJsonSync('server'));
             expect(datasources).to.not.include('ds-object-storage');
             var pkg = fs.readFileSync(
-              path.join(SANDBOX, 'test-app', 'package.json'), 'UTF-8');
+              path.join(SANDBOX, 'test-app', 'package.json'), 'UTF-8'
+            );
             pkg = JSON.parse(pkg);
             // eslint-disable-next-line no-unused-expressions
             expect(pkg.dependencies['loopback-component-storage']).to.exist;

--- a/test/export-api-def.test.js
+++ b/test/export-api-def.test.js
@@ -39,7 +39,8 @@ describe('loopback:export-api-def generator', function() {
       function(err, model) {
         test.Model = model;
         done(err);
-      });
+      }
+    );
   });
 
   before(function installDependencies(done) {
@@ -67,7 +68,8 @@ describe('loopback:export-api-def generator', function() {
       expect(content).to.have.property('basePath');
       expect(content).to.have.property('info');
       expect(content.info).to.have.property(
-        'title', 'test-app');
+        'title', 'test-app'
+      );
       expect(content).to.have.property('tags');
       expect(content.consumes).to.have.members([
         'application/json',
@@ -96,7 +98,8 @@ describe('loopback:export-api-def generator', function() {
       expect(content).to.have.property('basePath');
       expect(content).to.have.property('info');
       expect(content.info).to.have.property(
-        'title', 'test-app');
+        'title', 'test-app'
+      );
       expect(content).to.have.property('tags');
       expect(content.consumes).to.have.members([
         'application/json',

--- a/test/fixtures/help-texts/loopback_app_help.txt
+++ b/test/fixtures/help-texts/loopback_app_help.txt
@@ -7,6 +7,8 @@ Options:
         --skip-install     # Do not install npm dependencies                         Default: false
         --skip-next-steps  # Do not print "next steps" info
         --explorer         # Add Loopback Explorer to the project (true by default)
+        --loopbackVersion  # Select the LoopBack version
+        --template         # Set up the LoopBack application template
         --bluemix          # Set up as a Bluemix app
 
 Arguments:

--- a/test/middleware.test.js
+++ b/test/middleware.test.js
@@ -71,7 +71,8 @@ describe('loopback:middleware generator', function() {
       });
 
       var builtinSources = Object.keys(
-        readMiddlewaresJsonSync('server').routes);
+        readMiddlewaresJsonSync('server').routes
+      );
       modelGen.run(function() {
         var newSources = Object.keys(readMiddlewaresJsonSync('server').routes);
         var expectedSources = builtinSources.concat(['my-middleware-3']);
@@ -91,10 +92,12 @@ describe('loopback:middleware generator', function() {
     });
 
     var builtinSources = Object.keys(
-      readMiddlewaresJsonSync('server')['routes:after'] || {});
+      readMiddlewaresJsonSync('server')['routes:after'] || {}
+    );
     modelGen.run(function() {
       var newSources = Object.keys(
-        readMiddlewaresJsonSync('server')['routes:after']);
+        readMiddlewaresJsonSync('server')['routes:after']
+      );
       var expectedSources = builtinSources.concat(['my-middleware-4']);
       expect(newSources).to.have.members(expectedSources);
       done();

--- a/test/property.test.js
+++ b/test/property.test.js
@@ -32,7 +32,8 @@ describe('loopback:property generator', function() {
       function(err, model) {
         test.Model = model;
         done(err);
-      });
+      }
+    );
   });
 
   it('creates model with non required default', function(done) {

--- a/test/relations.test.js
+++ b/test/relations.test.js
@@ -40,8 +40,10 @@ describe('loopback:relation generator', function() {
           },
           function(err) {
             done(err);
-          });
-      });
+          }
+        );
+      }
+    );
   });
 
   it('adds an entry to common/models/{name}.json', function(done) {
@@ -221,8 +223,7 @@ describe('loopback:relation generator', function() {
         expect(Object.keys(relations)).to.include('part');
         done();
       });
-    }
-  );
+    });
 
   // requires generator-yeoman v0.17
   it('provides default property name based on target model for hasMany',
@@ -239,8 +240,7 @@ describe('loopback:relation generator', function() {
         expect(Object.keys(relations)).to.include('parts');
         done();
       });
-    }
-  );
+    });
 
   function givenRelationGenerator() {
     var name = 'loopback:relation';

--- a/test/remote-method.test.js
+++ b/test/remote-method.test.js
@@ -32,7 +32,8 @@ describe('loopback:remote-method generator', function() {
       function(err, model) {
         test.Model = model;
         done(err);
-      });
+      }
+    );
   });
 
   it('adds an entry to common/models/{name}.json', function(done) {

--- a/test/soap.test.js
+++ b/test/soap.test.js
@@ -25,8 +25,6 @@ describe('loopback:soap tests', function() {
       var modelGen = givenDataSourceGenerator();
       helpers.mockPrompt(modelGen, {
         name: 'soapds',
-        customConnector: '', // temporary workaround for
-        // https://github.com/yeoman/generator/issues/600
         connector: 'soap',
         url: 'http://www.webservicex.net/stockquote.asmx',
         wsdl: path.join(__dirname, 'soap/stockquote.wsdl'),
@@ -75,7 +73,7 @@ describe('loopback:soap tests', function() {
       });
   });
 
-  describe('periodic table wsdl', function() {
+  describe('calculator wsdl', function() {
     beforeEach(common.resetWorkspace);
 
     beforeEach(function createSandbox(done) {
@@ -90,11 +88,9 @@ describe('loopback:soap tests', function() {
       var modelGen = givenDataSourceGenerator();
       helpers.mockPrompt(modelGen, {
         name: 'soapds',
-        customConnector: '', // temporary workaround for
-        // https://github.com/yeoman/generator/issues/600
         connector: 'soap',
-        url: 'http://www.webservicex.net/periodictable.asmx',
-        wsdl: 'http://www.webservicex.net/periodictable.asmx?WSDL',
+        url: 'http://www.dneonline.com/calculator.asmx',
+        wsdl: 'http://www.dneonline.com/calculator.asmx?WSDL',
         remotingEnabled: true,
         installConnector: false,
       });
@@ -103,44 +99,70 @@ describe('loopback:soap tests', function() {
       });
     });
 
-    it('periodic table wsdl',
+    it('calculator wsdl',
       function(done) {
         var modelGen = givenModelGenerator();
         helpers.mockPrompt(modelGen, {
           dataSource: 'soapds',
-          url: 'http://www.webservicex.net/periodictable.asmx?WSDL',
-          service: 'periodictable',
-          binding: 'periodictableSoap',
-          operations: ['GetAtomicNumber', 'GetAtomicWeight'],
+          url: 'http://www.dneonline.com/calculator.asmx?WSDL',
+          service: 'Calculator',
+          binding: 'CalculatorSoap',
+          operations: ['Add', 'Divide', 'Multiply', 'Subtract'],
         });
-        // this runs command  loopback:soap command with mock up /test/soap/stockquote.wsdl as input from command prompt
+
         modelGen.run(function() {
-          var content = readModelJsonSync('get-atomic-weight');
+          var content = readModelJsonSync('add');
           expect(content).to.not.have.property('public');
           expect(content).to.have.property('properties');
-          expect(content.properties.ElementName.type).to.eql('string');
+          expect(content.properties.intA.type).to.eql('number');
+          expect(content.properties.intB.type).to.eql('number');
           expect(content).to.have.property('excludeBaseProperties');
           var expectedExcludeProps = ['id'];
           expect(content.excludeBaseProperties).
             to.deep.equal(expectedExcludeProps);
 
-          content = readModelJsonSync('get-atomic-weight-response');
-          expect(content.properties.GetAtomicWeightResult.type).to.eql('string'); // eslint-disable-line max-len
+          content = readModelJsonSync('add-response');
+          expect(content.properties.AddResult.type).to.eql('number');
 
-          content = readModelJsonSync('get-atomic-number');
+          content = readModelJsonSync('divide');
           expect(content).to.not.have.property('public');
           expect(content).to.have.property('properties');
-          expect(content.properties.ElementName.type).to.eql('string');
+          expect(content.properties.intA.type).to.eql('number');
+          expect(content.properties.intB.type).to.eql('number');
 
-          content = readModelJsonSync('get-atomic-number-response');
-          expect(content.properties.GetAtomicNumberResult.type).to.eql('string'); // eslint-disable-line max-len
+          content = readModelJsonSync('divide-response');
+          expect(content.properties.DivideResult.type).to.eql('number');
+
+          content = readModelJsonSync('multiply');
+          expect(content).to.not.have.property('public');
+          expect(content).to.have.property('properties');
+          expect(content.properties.intA.type).to.eql('number');
+          expect(content.properties.intB.type).to.eql('number');
+
+          content = readModelJsonSync('multiply-response');
+          expect(content.properties.MultiplyResult.type).to.eql('number');
+
+          content = readModelJsonSync('subtract');
+          expect(content).to.not.have.property('public');
+          expect(content).to.have.property('properties');
+          expect(content.properties.intA.type).to.eql('number');
+          expect(content.properties.intB.type).to.eql('number');
+
+          content = readModelJsonSync('subtract-response');
+          expect(content.properties.SubtractResult.type).to.eql('number');
 
           var modelConfig = readModelConfigSync('server');
-          expect(modelConfig).to.have.property('GetAtomicWeight');
-          expect(modelConfig.GetAtomicWeight).to.have.property('public', true);
+          expect(modelConfig).to.have.property('Add');
+          expect(modelConfig.Add).to.have.property('public', true);
 
-          expect(modelConfig).to.have.property('GetAtomicNumber');
-          expect(modelConfig.GetAtomicNumber).to.have.property('public', true);
+          expect(modelConfig).to.have.property('Divide');
+          expect(modelConfig.Divide).to.have.property('public', true);
+
+          expect(modelConfig).to.have.property('Multiply');
+          expect(modelConfig.Multiply).to.have.property('public', true);
+
+          expect(modelConfig).to.have.property('Subtract');
+          expect(modelConfig.Subtract).to.have.property('public', true);
 
           done();
         });
@@ -159,8 +181,6 @@ describe('loopback:soap tests', function() {
       var modelGen = givenDataSourceGenerator();
       helpers.mockPrompt(modelGen, {
         name: 'soapds',
-        customConnector: '', // temporary workaround for
-        // https://github.com/yeoman/generator/issues/600
         connector: 'soap',
         url: 'http://localhost:15099/rpc_Literal_testing',
         wsdl: path.join(__dirname, 'soap/special_char_test.wsdl'),
@@ -182,7 +202,7 @@ describe('loopback:soap tests', function() {
           operations: ['myMethod'],
         });
 
-        // this runs command  loopback:soap command with mock up /test/soap/stockquote.wsdl as input from command prompt
+        // this runs command  loopback:soap command with mock up /test/soap/special_char_test.wsdl as input from command prompt
         modelGen.run(function() {
           var content = readAPIFileSync('soap-rpc-literal-test-2-0-binding');
           done();

--- a/test/zosconnectee.test.js
+++ b/test/zosconnectee.test.js
@@ -84,8 +84,7 @@ describe('loopback:zosconnectee generator', function() {
           'zosconnectee_template.json');
         done();
       });
-    }
-  );
+    });
 
   function givenDataSourceGenerator(dsArgs, _dsPath) {
     var dsPath = _dsPath || '../../datasource';

--- a/zosconnectee/index.js
+++ b/zosconnectee/index.js
@@ -42,7 +42,8 @@ module.exports = generators.Base.extend({
         that.datasources = tempList;
         if (that.datasources.length == 0) {
           that.env.error(
-            'Define a zosconnectee datasource before running this command');
+            'Define a zosconnectee datasource before running this command'
+          );
         }
         done(); // End the sync
       }

